tests/st/calicoctl/test_default_pools.py

```diff
@@ -0,0 +1,150 @@
+        cls.hosts = []
+        cls.host = DockerHost("host",
+                              additional_docker_options=ADDITIONAL_DOCKER_OPTIONS,
+                              start_calico=False)
+        cls.hosts.append(cls.host)
+
+    def setUp(self):
+        try:
+            self.host.execute("docker rm -f calico-node")
+        except CommandExecError:
+            # Presumably calico-node wasn't running
+            pass
+        wipe_etcd(get_ip())
+
+    @parameterized.expand([
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/27",  None),    # Too small
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/32",  None),    # Too small, but legal
+        (False, "CALICO_IPV4POOL_CIDR", "10.0.0.0/33",  None),    # Impossible CIDR
+        (False, "CALICO_IPV4POOL_CIDR", "256.0.0.0/24", None),    # Invalid IP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24",  None),    # Typical non-default pool
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/26",  None),    # Smallest legal pool
+        (True,  "CALICO_IPV6POOL_CIDR", "fd00::/122",   None),    # Smallest legal pool
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/123",   None),    # Too small
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/128",   None),    # Too small, but legal
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/129",   None),    # Impossible CIDR
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24",  "true"),  # Typical non-default pool, IPIP
+        (True,  "CALICO_IPV4POOL_CIDR", "10.0.0.0/24", "false"),  # Typical pool, explicitly no IPIP
+        (False, "CALICO_IPV6POOL_CIDR", "fd00::/122",   "true"),  # IPv6 - IPIP not permitted
+        (True,  "CALICO_IPV4POOL_CIDR", "0.0.0.0/0",    None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "::/0",         None),    # Valid, but not sensible
+        (True,  "CALICO_IPV6POOL_CIDR", "fd80::0:0/120", None),   # Valid, but non-canonical form
+        (False, "CALICO_IPV6POOL_CIDR", "1.2.3.4/24",    None),   # Wrong type
+        (False, "CALICO_IPV4POOL_CIDR", "fd00::/24",     None),   # Wrong type
+        (True,  "CALICO_IPV6POOL_CIDR", "::0:a:b:c:d:e:0/120", None),  # Valid, non-canonical form
+        (False, "CALICO_IPV4POOL_CIDR", "1.2/16",       None),    # Valid, unusual form
+    ])
+    def test_default_pools(self, success_expected, param, value, ipip):
+        """
+        Test that the various options for default pools work correctly
+        """
+        # Get command line for starting docker
+        output = self.host.calicoctl("node run --dryrun")
+        base_command = output.split('\n')[-4].rstrip()
+
+        # Modify command line to add the options we want to test
+        env_index = base_command.find("-e")
+        command = base_command[:env_index] + "-e %s=%s " % (param, value) + base_command[env_index:]
+        if ipip is not None:
+            command = command[:env_index] + \
+                      "-e CALICO_IPIP_ENABLED=%s " % ipip + \
```

Comment:
The changes that this PR is waiting on also change the name of this variable to CALICO_IPV4POOL_IPIP with possible values of <empty string>, "off", "always", and "cross-subnet".  Just pointing that out but you probably already know changes were coming.